src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -216,14 +217,14 @@ export default class DefaultDeviceController
   }
 
   async startAudioInput(device: AudioInputDevice): Promise<MediaStream | undefined> {
+    return (await this.audioInputQueue.add(() => this.startAudioInputInner(device))) as MediaStream;
```

Comment:
A better way would be to use a TypeScript generic instead of type casting.

```
add<T>(func: () => Promise<T>): Promise<T> {
```

Reply:
Thank you very much. I tried <T> this morning but probably did it wrong as it did not work :p.